src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp

```diff
@@ -171,6 +178,7 @@ namespace Quantum
             }
         }
 
+        // To do: remove after the `function DumpMachine<'T> (location : 'T) : Unit` is implemented.

```

Comment:
This might not need to be removed; as discussed earlier, host programs can make use of structured diagnostics (e.g.: the state vector represented as a vector of complex numbers, rather than as a human-readable string).

Reply:
The `GetState()` is only called by `CFullstateSimulator::DumpState()`, which is never called (not to be confused with the C#'s `DumpState()`).
The `DumpMachine()` and `DumpRegister()` added in this PR, provide the same functionality. 
After this PR I plan to remove `GetState()` and `DumpState()`.

Reply:
#636